libs/auth/src/angular/lock/lock-component.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Observable } from "rxjs";
+
+import { UserId } from "@bitwarden/common/types/guid";
+
+export enum BiometricsDisableReason {
+  NotSupportedOnOperatingSystem = "NotSupportedOnOperatingSystem",
+  EncryptedKeysUnavailable = "BiometricsEncryptedKeysUnavailable",
+  SystemBiometricsUnavailable = "SystemBiometricsUnavailable",
+}
+
+// The options here should match the top level properties of the UnlockOptions type.
+export enum UnlockOption {
+  MasterPassword = "masterPassword",
+  Pin = "pin",
+  Biometrics = "biometrics",
+}
```

Comment:
You can immortalize the rule of it matching all top level properties in UnlockOptions and avoid using an enum in one go by doing something like:

```suggestion
export const UnlockOption = Object.freeze({
  MasterPassword: "masterPassword",
  Pin: "pin",
  Biometrics: "biometrics",
}) satisfies { [Prop in keyof UnlockOptions as Capitalize<Prop>]: Prop }
```

Reply:
Thanks for this suggestion and working through the implementation with me!